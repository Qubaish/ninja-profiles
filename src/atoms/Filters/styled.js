import styled from 'styled-components';

export const StyledFilterWrapper = styled.div`
    min-height: 5rem;
    background: #fff;
    margin-bottom: 2rem;
    padding: 1rem;
    background-color: #FCFDFF;
    border-radius: 0.3rem;
    display: flex;
    justify-content: space-between;
`;

export const StyledInput = styled.input`
    display: block;
    margin: 0;
    padding: 0.8rem 0.8rem;
    color: inherit;
    height: 50px;
    width: 48%;
    font-family: inherit;
    font-size: 1rem;
    font-weight: inherit;
    line-height: 1.8;
    border: 1px solid #dcdde0;
    border-radius: 0.4rem;
    transition: box-shadow 300ms;
    box-sizing: border-box;
    :focus {
        outline: none;
        box-shadow: 0 3px 12px 0 rgba(0, 0, 0, 0.5);
    }
`;

export const StyledSelect = styled.select`
    height: 5rem;
    width: 48%;
    border: none;
    padding: 1rem;
    border: 1px solid #dcdde0;
    border-radius: 0.3rem;
    box-shadow: 1px 2px 3px rgba(220,221,224,0.6);
    :focus {
        outline: none;
    }
`;