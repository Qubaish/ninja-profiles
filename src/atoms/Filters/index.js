import React from 'react';
import PropTypes from 'prop-types';
import { StyledFilterWrapper, StyledInput, StyledSelect} from './styled';

export default function Filters({cities, onNameChange, onLocationSelect}){
    return (
        <StyledFilterWrapper>
            <StyledInput data-testid="input" placeholder="Name e.g John" name="name" onChange={onNameChange}/>
            <StyledSelect data-testid="select" name="city" onChange={onLocationSelect}>
                <option value="">Filter by Offices</option>
                {cities.map((c, i) => (<option value={c} key={i}>{c}</option>))}
            </StyledSelect>
        </StyledFilterWrapper>
    );
}

Filters.propTypes = {
    cities: PropTypes.array,
    onNameChange: PropTypes.func,
    onLocationSelect: PropTypes.func,
};