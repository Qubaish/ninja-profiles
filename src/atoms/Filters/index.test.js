import React from 'react';
import { render } from '@testing-library/react';
import Filters from './index';

describe('<Filters/>', () => {
    it('checking snapshot', () => {
        const { asFragment } = render(<Filters cities={['lund']} onNameChange={jest.fn()} onLocationSelect={jest.fn()}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('shoul render with correct data', () => {
        const { getByTestId } = render(<Filters cities={['lund']} onNameChange={jest.fn()} onLocationSelect={jest.fn()}/>);
        const elem = getByTestId('input');
        const selectElem = getByTestId('select');
        expect(elem.name).toBe('name');
        console.log(selectElem.children);
        expect(selectElem.children).toHaveLength(2);
        expect(selectElem.children[1].innerHTML).toBe('lund');
    });
})