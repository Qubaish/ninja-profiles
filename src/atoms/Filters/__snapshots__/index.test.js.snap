// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters/> checking snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm idaAJJ"
  >
    <input
      class="sc-AxirZ beswKk"
      data-testid="input"
      name="name"
      placeholder="Name e.g John"
    />
    <select
      class="sc-AxiKw pHQxu"
      data-testid="select"
      name="city"
    >
      <option
        value=""
      >
        Filter by Offices
      </option>
      <option
        value="lund"
      >
        lund
      </option>
    </select>
  </div>
</DocumentFragment>
`;
